sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -288,8 +288,17 @@ private void CloseLink(ReceivingAmqpLink link)
             IEnumerable<string> lockTokens,
             TimeSpan timeout)
         {
-            Guid[] lockTokenGuids = lockTokens.Select(token => new Guid(token)).ToArray();
-            if (lockTokenGuids.Any(lockToken => _requestResponseLockedMessages.Contains(lockToken)))
+            bool requestResponse = false;
+            var lockTokenGuids = lockTokens.Select(token => new Guid(token)).ToArray();
```

Comment:
Why not get rid of this piece of linq too?

Reply:
I did an attempt to do it by creating an array of guid. I then needed to ask for the count on the enumerable which lead to less allocations as the original version but worse performance